<!DOCTYPE html><html><head><meta charSet="utf-8"/><meta name="viewport" content="width=device-width"/><title>My new cool app</title><meta name="next-head-count" content="3"/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/_next/static/chunks/polyfills-c67a75d1b6f99dc8.js"></script><script src="/_next/static/chunks/webpack-d38be8d96a62f950.js" defer=""></script><script src="/_next/static/chunks/framework-3b5a00d5d7e8d93b.js" defer=""></script><script src="/_next/static/chunks/main-1b5205ad35b92557.js" defer=""></script><script src="/_next/static/chunks/pages/_app-72ed113f22a36849.js" defer=""></script><script src="/_next/static/chunks/675-44addd24cc62f3a3.js" defer=""></script><script src="/_next/static/chunks/pages/blog/%5Bslug%5D-6a5b6173e98f113c.js" defer=""></script><script src="/_next/static/CX8-OElU8TqTkW2zxyeBX/_buildManifest.js" defer=""></script><script src="/_next/static/CX8-OElU8TqTkW2zxyeBX/_ssgManifest.js" defer=""></script><style data-styled="" data-styled-version="5.3.6">.iBQlkL{background:coral;height:60px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
@media (max-width:920px){.iBQlkL ul{display:none;}}/*!sc*/
data-styled.g1[id="sc-7295e4c-0"]{content:"iBQlkL,"}/*!sc*/
.bDMZDz{display:none;font-size:22px;margin-left:auto;}/*!sc*/
@media (max-width:920px){.bDMZDz{display:block;}}/*!sc*/
data-styled.g2[id="sc-7295e4c-1"]{content:"bDMZDz,"}/*!sc*/
html,body{height:100%;margin:0;}/*!sc*/
body{background:#fff;}/*!sc*/
html{box-sizing:border-box;font-size:16px;}/*!sc*/
*,*:before,*:after{box-sizing:inherit;}/*!sc*/
body,h1,h2,h3,h4,h5,h6,p,ol,ul{margin:0;padding:0;font-weight:normal;}/*!sc*/
ol,ul{list-style:none;}/*!sc*/
img,video{max-width:100%;height:auto;}/*!sc*/
data-styled.g3[id="sc-global-eYmBoQ1"]{content:"sc-global-eYmBoQ1,"}/*!sc*/
.ufdhg{max-width:800px;margin:0 auto;}/*!sc*/
data-styled.g4[id="sc-dab4a0e4-0"]{content:"ufdhg,"}/*!sc*/
</style></head><body><div id="__next"><nav class="sc-7295e4c-0 iBQlkL"><ul><a href="/blog">Blog</a></ul><button class="sc-7295e4c-1 bDMZDz">X</button></nav><main><div class="sc-dab4a0e4-0 ufdhg"><h1>NetworkX in Orange</h1><p><a href="http://networkx.lanl.gov/">NetworkX</a> – a popular open-source python library for network analysis has finally found its way into Orange. It is now used as a base class for network representation in all Orange modules and widgets. By that, we offered to the widespread network community a fruitful and fun way to visualize and explore networks, using their existing NetworkX scripts. It has never been easier to combine network analysis and visualization with existing machine learning and data discovery methods.</p>
<p>Complete documentation is available in the <a href="/doc/orange25/Orange.network.html">Orange network headquarters</a>. For a brief overview, take a look at the following example. Let us suppose we would like to analyse the data about patients, having one of two types of leukemia. So, we have a <a href="http://blog.biolab.si/wp-content/uploads/2011/07/29/leukemiagsea.tab">data set</a> with 72 patient, 4600+ gene expressions and a class variable. We also have a vast <a href="/blog/2011-07-networkx-in-orange/genes_biofuncttar.gz">network of human genes</a>, connected if they share a biological function. What we would like to examine is a sub-network with only several hundred most expressed genes from the data set. To show off a bit, we will also use the <a href="/addons/">Orange Bioinformatics add-on</a>. Here is how we do it:</p>
<pre><code>    import Orange
    import obiExpression

    # load leukemia data set
    table = Orange.data.Table(&quot;/media/Ox/Projects_Archive/res/BIO/leukemia/leukemiaGSEA.tab&quot;)

    useAttributeLabels = False
    ttest = obiExpression.ExpressionSignificance_TTest(table, useAttributeLabels)

    target = [table.domain.classVar(0), table.domain.classVar(1)]

    # test for significantly expressed genes
    score = ttest(target = target)

    # each gene is scored (t-test, p-value)
    print score[0]
    &gt;&gt;&gt; (FloatVariable &#x27;HIST1H4C&#x27;, (1.8377179790830149, 0.07034778767062116))

    # sort by p-value
    from operator import itemgetter
    score.sort(key=lambda s: s[1][1])

    # select 200 genes with the lowest p-value
    important_genes = [gene_var.name for gene_var, s in score[:200]]

    # read the gene network (5000+ genes, dense network)
    G = Orange.network.readwrite.read(&#x27;genes_biofunct.gpickle&#x27;)

    items = G.items().filter_bool({&#x27;gene&#x27;: important_genes})
    indices = [i for i, present in enumerate(items) if present]

    # build a subraph of 200 most expressed genes
    G_sub = G.subgraph(indices)
</code></pre>
<p>In addition to the power of scripting environment, we also get the benefits of visual data exploration with Orange widgets. However, network widgets are currently under heavy development, so expect some bugs if you dare to try them. Coding should be finished in a month or two, check the blog for progress updates. Here is how to open the network in Nx Explorer widget:</p>
<pre><code>    import sys
    import PyQt4

    # must have OWNxExplorer in python path!
    import OWNxExplorer

    app=PyQt4.QtGui.QApplication(sys.argv)
    ow=OWNxExplorer.OWNxExplorer()
    ow.show()

    # set the network
    ow.set_graph(G_sub)
    app.exec_()
</code></pre></div></main></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{"frontmatter":{"author":"BIOLAB","date":"2011-07-29 18:56:00+00:00","draft":false,"title":"NetworkX in Orange","type":"blog","blog":["analysis","network","networkx","visualization"]},"content":{"compiledSource":"/*@jsxRuntime automatic @jsxImportSource react*/\nconst {Fragment: _Fragment, jsx: _jsx, jsxs: _jsxs} = arguments[0];\nconst {useMDXComponents: _provideComponents} = arguments[0];\nfunction _createMdxContent(props) {\n  const _components = Object.assign({\n    p: \"p\",\n    a: \"a\",\n    pre: \"pre\",\n    code: \"code\"\n  }, _provideComponents(), props.components);\n  return _jsxs(_Fragment, {\n    children: [_jsxs(_components.p, {\n      children: [_jsx(_components.a, {\n        href: \"http://networkx.lanl.gov/\",\n        children: \"NetworkX\"\n      }), \" – a popular open-source python library for network analysis has finally found its way into Orange. It is now used as a base class for network representation in all Orange modules and widgets. By that, we offered to the widespread network community a fruitful and fun way to visualize and explore networks, using their existing NetworkX scripts. It has never been easier to combine network analysis and visualization with existing machine learning and data discovery methods.\"]\n    }), \"\\n\", _jsxs(_components.p, {\n      children: [\"Complete documentation is available in the \", _jsx(_components.a, {\n        href: \"/doc/orange25/Orange.network.html\",\n        children: \"Orange network headquarters\"\n      }), \". For a brief overview, take a look at the following example. Let us suppose we would like to analyse the data about patients, having one of two types of leukemia. So, we have a \", _jsx(_components.a, {\n        href: \"http://blog.biolab.si/wp-content/uploads/2011/07/29/leukemiagsea.tab\",\n        children: \"data set\"\n      }), \" with 72 patient, 4600+ gene expressions and a class variable. We also have a vast \", _jsx(_components.a, {\n        href: \"/blog/2011-07-networkx-in-orange/genes_biofuncttar.gz\",\n        children: \"network of human genes\"\n      }), \", connected if they share a biological function. What we would like to examine is a sub-network with only several hundred most expressed genes from the data set. To show off a bit, we will also use the \", _jsx(_components.a, {\n        href: \"/addons/\",\n        children: \"Orange Bioinformatics add-on\"\n      }), \". Here is how we do it:\"]\n    }), \"\\n\", _jsx(_components.pre, {\n      children: _jsx(_components.code, {\n        children: \"    import Orange\\n    import obiExpression\\n\\n    # load leukemia data set\\n    table = Orange.data.Table(\\\"/media/Ox/Projects_Archive/res/BIO/leukemia/leukemiaGSEA.tab\\\")\\n\\n    useAttributeLabels = False\\n    ttest = obiExpression.ExpressionSignificance_TTest(table, useAttributeLabels)\\n\\n    target = [table.domain.classVar(0), table.domain.classVar(1)]\\n\\n    # test for significantly expressed genes\\n    score = ttest(target = target)\\n\\n    # each gene is scored (t-test, p-value)\\n    print score[0]\\n    \u003e\u003e\u003e (FloatVariable 'HIST1H4C', (1.8377179790830149, 0.07034778767062116))\\n\\n    # sort by p-value\\n    from operator import itemgetter\\n    score.sort(key=lambda s: s[1][1])\\n\\n    # select 200 genes with the lowest p-value\\n    important_genes = [gene_var.name for gene_var, s in score[:200]]\\n\\n    # read the gene network (5000+ genes, dense network)\\n    G = Orange.network.readwrite.read('genes_biofunct.gpickle')\\n\\n    items = G.items().filter_bool({'gene': important_genes})\\n    indices = [i for i, present in enumerate(items) if present]\\n\\n    # build a subraph of 200 most expressed genes\\n    G_sub = G.subgraph(indices)\\n\"\n      })\n    }), \"\\n\", _jsx(_components.p, {\n      children: \"In addition to the power of scripting environment, we also get the benefits of visual data exploration with Orange widgets. However, network widgets are currently under heavy development, so expect some bugs if you dare to try them. Coding should be finished in a month or two, check the blog for progress updates. Here is how to open the network in Nx Explorer widget:\"\n    }), \"\\n\", _jsx(_components.pre, {\n      children: _jsx(_components.code, {\n        children: \"    import sys\\n    import PyQt4\\n\\n    # must have OWNxExplorer in python path!\\n    import OWNxExplorer\\n\\n    app=PyQt4.QtGui.QApplication(sys.argv)\\n    ow=OWNxExplorer.OWNxExplorer()\\n    ow.show()\\n\\n    # set the network\\n    ow.set_graph(G_sub)\\n    app.exec_()\\n\"\n      })\n    })]\n  });\n}\nfunction MDXContent(props = {}) {\n  const {wrapper: MDXLayout} = Object.assign({}, _provideComponents(), props.components);\n  return MDXLayout ? _jsx(MDXLayout, Object.assign({}, props, {\n    children: _jsx(_createMdxContent, props)\n  })) : _createMdxContent(props);\n}\nreturn {\n  default: MDXContent\n};\n","frontmatter":{},"scope":{}}},"__N_SSG":true},"page":"/blog/[slug]","query":{"slug":"networkx-in-orange"},"buildId":"CX8-OElU8TqTkW2zxyeBX","isFallback":false,"gsp":true,"scriptLoader":[]}</script></body></html>